components/places/src/bookmark_sync/incoming.rs

```diff
@@ -390,36 +466,112 @@ impl<'a> IncomingApplicator<'a> {
     }
 }
 
+fn unpack_id(key: &str, data: &JsonValue) -> Result<BookmarkRecordId> {
+    let id = &data[key].as_str();
```

Comment:
An `if let` might be cleaner here, like you did for tags above.

Reply:
Done!